src/main/java/Entities/Implementations/AbstractRecipeItem.java

```diff
@@ -0,0 +1,46 @@
+package Entities.Implementations;
+
+import Entities.Ingredient;
+import EntityInterfaces.RecipeItem;
+
+public abstract class AbstractRecipeItem implements RecipeItem {
+    public Ingredient ingredient;
+    public float amount;
+    public boolean optional = false;
```

Comment:
These should be protected not public. We do not want outside classes to access these.